rubrik_cdm/data_management.py

```diff
@@ -1996,28 +1996,34 @@ def sql_db_export(self, db_name, date, time, sql_instance=None, sql_host=None, t
 
             return self.post('v1', '/mssql/db/{}/export'.format(mssql_id), config, timeout)
 
-    def set_esx_subnets(self, esx_subnets=None, timeout=15):  # pylint: ignore
+    def set_esxi_subnets(self, esx_subnets=None, timeout=15):  # pylint: ignore
         """Sets the subnets that should be used to reach the ESXi hosts.
         Keyword Arguments:
-            esx_subnets {str} -- Preferred subnets used to reach the ESX hosts. The format should be the comma separated list without any spaces, for example, 192.168.2.10/24,10.255.0.2/16.
+            esx_subnets {list} -- Preferred subnets used to reach the ESX hosts. The format should be a list, for example ['192.168.2.10/24','10.255.0.2/16'].
             timeout {int} -- The number of seconds to wait to establish a connection with the Rubrik cluster before returning a timeout error. (default: {15})
         Returns:
             dict -- The full response of `PATCH /internal/vmware/config/set_esx_subnets`.
         """
+        self.log("set_esx_subnets: Getting the existing subnets used to reach the ESXi hosts")
+        subnets_data = self.get_esxi_subnets()
+        subnets = subnets_data['esxSubnets'].split(',')
 
         config = {}
         if esx_subnets is None:
```

Comment:
Let's also add a check to validate it's a list and return an appropriate error message if its not

Reply:
Good call, made the changes now